src/corehost/common/pal.h

```diff
@@ -180,16 +180,10 @@ namespace pal
 
 #endif
 
-#if defined(__APPLE__)
-    enum os_moniker_t
-    {
-        sierra = 16,
-        el_capitan = 15,
-    };
-
-    bool get_os_moniker(os_moniker_t* moniker);
-#endif
-
+    // Based upon https://github.com/dotnet/core-setup/blob/master/src/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs
+    pal::string_t get_current_os_rid_platform();
+    pal::string_t cleanup_quotes(pal::string_t stringToCleanup);
```

Comment:
Is it a problem that we define this method in the common "pal" header, but it is only used and implemented on unix?